app/serializers/category_detailed_serializer.rb

```diff
@@ -28,10 +28,6 @@ def is_uncategorized
     object.id == SiteSetting.uncategorized_category_id
   end
 
-  def include_is_uncategorized?
```

Comment:
I'm not sure why this method is being removed?

Reply:
Right! I meant to write something up about this, but I got distracted. Glad you called it out :)

The `is_uncategorized` property of the items in the `categories` array returned by `GET /categories.json` is marked in the openapi spec as `required` (https://github.com/discourse/discourse/blob/main/spec/requests/api/schemas/json/category_list_response.json#L200).

This `include_is_uncategorized?` method was causing the property to be elided when null, i.e. violating the openapi spec.

The `categories /categories.json get success response behaves like a JSON endpoint response body matches the documented response schema` *should* have caught this discrepancy, but it wasn't working! A structural problem in the openapi spec (`categories` items was an array instead of an object) caused it to effectively not validate the contents of the array.

So, when I fixed the structural problem with the openapi spec, the existing test (correctly!) called out that a required field was missing from the `categories.json` response payload. Telling the serializer to unilaterally include the property (by removing this method) fixed it.

I can't be 100% sure that this is the correct fix, from where I sit. The main alternative would be to remove `is_uncategorized` from the `required` list for that object in the openapi spec.

Reply:
Ah okay. Ya lets go ahead and remove is_uncategorized from the required list instead. This way we aren't deleting any methods.

Reply:
done!